go/processor/floria/state_contract.go

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strings"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	callValueTransferGas = tosca.Gas(9000)
+	createGas            = tosca.Gas(32000)
+	createDataGas        = tosca.Gas(200)
+	memoryGas            = tosca.Gas(3)
+	sStoreSetGasEIP2200  = tosca.Gas(20000)
+)
+
+// DriverAddress is the NodeDriver contract address
+var DriverAddress = tosca.Address(common.HexToAddress("0xd100a01e00000000000000000000000000000000"))
+
+// StateContractAddress is the EvmWriter pre-compiled contract address
+var StateContractAddress = tosca.Address(common.HexToAddress("0xd100ec0000000000000000000000000000000000"))
```

Comment:
Could those be private? Since those are not constants, they could be updated accidentally.